mapie/classification.py

```diff
@@ -658,95 +859,147 @@ def predict(
+            elif agg_scores == "mean":
+                y_pred_proba = np.mean(y_pred_proba_k, axis=0)
+                y_pred_proba = self._check_proba_normalized(
+                    y_pred_proba, axis=1
+                )
+            else:
+                raise ValueError("Invalid 'agg_scores' argument.")
+
+        # Estimate prediction sets
         n = self.n_samples_val_
         if alpha_ is None:
             return np.array(y_pred)
-        else:
 
+        else:
             # Choice of the quantile
             check_alpha_and_n_samples(alpha_, n)
             if self.method == "naive":
                 self.quantiles_ = 1 - alpha_
             else:
-                self.quantiles_ = np.stack([
-                    np.quantile(
-                        self.conformity_scores_,
-                        ((n + 1) * (1 - _alpha)) / n,
-                        interpolation="higher"
-                    ) for _alpha in alpha_
-                ])
+                if (cv == "prefit") or (agg_scores in ["mean"]):
+                    self.quantiles_ = np.stack([
+                        np.quantile(
+                            self.conformity_scores_,
+                            ((n + 1) * (1 - _alpha)) / n,
+                            interpolation="higher"
+                        ) for _alpha in alpha_
+                    ])
+                else:
+                    self.quantiles_ = (n + 1) * (1 - alpha_)
+
+            # Build prediction sets
             if self.method == "score":
-                prediction_sets = np.stack(
-                    [
-                        y_pred_proba > 1 - quantile
-                        for quantile in self.quantiles_
-                    ],
-                    axis=2,
-                )
-            elif self.method in ["cumulated_score", "naive"]:
-                # sort labels by decreasing probability
-                index_sorted = np.fliplr(np.argsort(y_pred_proba, axis=1))
-
-                # sort probabilities by decreasing order
-                y_pred_proba_sorted = np.take_along_axis(
-                    y_pred_proba, index_sorted, axis=1
-                )
-
-                # get sorted cumulated score
-                y_pred_proba_sorted_cumsum = np.cumsum(
-                    y_pred_proba_sorted, axis=1
-                )
+                if (cv == "prefit") or (agg_scores == "mean"):
+                    prediction_sets = np.stack(
+                        [
+                            y_pred_proba > 1 - quantile
+                            for quantile in self.quantiles_
+                        ],
+                        axis=2,
+                    )
+                else:
+                    y_comp = (
+                        1 - y_pred_proba < self.conformity_scores_[:, None]
+                    ).sum(axis=0)
+                    prediction_sets = np.stack(
+                        [
+                            y_comp > (n + 1 - quantile_)
```

Comment:
The lower bound seems odds, maybe we should discuss about it ?

Reply:
Similar to the split-conformal method in which the probabilities are compared with `1-quantile` (see line 930), we compare the number of times the inequality is verified with `(n + 1 - quantile)` .

Reply:
Changed to `y_comp > alpha2 * (n - 1)`